### PR TITLE
Convert CSVDiff to use SchemaDiff backend for diffing

### LINKED ISSUE #23522: Strange behavior with CSVDiff with abs_zero
## Bug Description
When comparing two valid CSV files (one produced by running the test, the other representing the gold file), the test harness may print out something like

```
test:dummy.test: ERROR: Traceback (most recent call last):
test:dummy.test:   File "/home/anovak/moose/python/mooseutils/csvdiff.py", line 399, in <module>
test:dummy.test:     messages = csv_differ.diff()
test:dummy.test:   File "/home/anovak/moose/python/mooseutils/csvdiff.py", line 327, in diff
test:dummy.test:     table2[key].index(val2),
test:dummy.test: ValueError: 0 is not in list
```

when using `abs_zero` (but disappears when not using `abs_zero`). I think this may be happening if one of the numbers to be compared is above `abs_zero`, but the other is below it.

## Steps to Reproduce

I have attached a `test.i`, with a bunch of postprocessor values. I copied these verbatim from an actual Cardinal test we're seeing this problem with, which is why there's so many (wasn't sure which one was the culprit). Accompanying this is our actual `test_out.csv`. 

Run `./run_tests --re=dummy` to get the error. If you comment out `abs_zero`, the error goes away and instead shows the typical postprocessor diff summary.

## Impact
If this error is a manifestation of something wrong with our test itself, it would be helpful to understand what that may be. Otherwise, this is preventing a new feature being merged in Cardinal.

tag @anshchaube

### PR DESCRIPTION
**Pull request by socratesgorilla**
Closes #23522

### CODE REVIEW

**Comment by socratesgorilla** [comment]
Officially open for real review. Current failures are due to weird outputs in modules unrelated.

**Comment by loganharbour** [comment]
Taking a look at this. Just invalidated tests

**Comment by GiudGiud** [comment]
```
stochastic_tools/test:vectorpostprocessors/stochastic_results_complete_history.data/time:   File "/opt/civet0/build/moose/python/TestHarness/testers/SchemaDiff.py", line 116, in check_values
stochastic_tools/test:vectorpostprocessors/stochastic_results_complete_history.data/time:     elif abs(x-y)/max(abs(x), abs(y)) > self.rel_err:
stochastic_tools/test:vectorpostprocessors/stochastic_results_complete_history.data/time: ZeroDivisionError: float division by zero
```
seems there s a relative diff on something that is equal to 0.
Should not be too hard to track / fix. The schemadiff should be able to not die on this.

**Comment by socratesgorilla** [comment]
> ```
> stochastic_tools/test:vectorpostprocessors/stochastic_results_complete_history.data/time:   File "/opt/civet0/build/moose/python/TestHarness/testers/SchemaDiff.py", line 116, in check_values
> stochastic_tools/test:vectorpostprocessors/stochastic_results_complete_history.data/time:     elif abs(x-y)/max(abs(x), abs(y)) > self.rel_err:
> stochastic_tools/test:vectorpostprocessors/stochastic_results_complete_history.data/time: ZeroDivisionError: float division by zero
> ```
> 
> seems there s a relative diff on something that is equal to 0. Should not be too hard to track / fix. The schemadiff should be able to not die on this.

The issue here is that apparently a bunch of stochastic tools tests that were formally outputting NaN are outputting 0, and Cody said that that should be looked into rather than changing the schemadiff to omit the differences

**Comment by GiudGiud** [comment]
@zachmprince do you know what is going on here

File: test/tests/test_harness/csvdiff_tests
Comment (loganharbour) [inline test/tests/test_harness/csvdiff_tests:1]: Can you not `git mv` this instead? it's very hard to tell whether or not you've removed tests or just moved them

**Review by loganharbour — CHANGES_REQUESTED** [review]

File: python/TestHarness/schedulers/RunParallel.py [hunk 1]
```diff
@@ -80,7 +80,7 @@ def run(self, job):
                 self.setSuccessfulMessage(tester)
         except Exception as e:
             output += 'Python exception encountered:\n\n' + traceback.format_exc()
-            tester.setStatus(tester.error, 'PYTHON EXCEPTION')
```
[the remark this answers is not in the document]
Comment (socratesgorilla) [reply python/TestHarness/schedulers/RunParallel.py:83]: yeah, this is a legitimate fix

File: python/TestHarness/tests/test_Diff.py [hunk 2]
```diff
@@ -20,9 +20,7 @@ def testDiffs(self):
 
         e = cm.exception
         self.assertRegex(e.output.decode('utf-8'), r'test_harness\.exodiff.*?FAILED \(EXODIFF\)')
-        self.assertRegex(e.output.decode('utf-8'), r'test_harness\.csvdiff.*?FAILED \(CSVDIFF\)')
```
[the remark this answers is not in the document]
Comment (socratesgorilla) [reply python/TestHarness/tests/test_Diff.py:23]: Superfluous with the new schemadiff tests

File: python/doc/content/python/CSVDiff.md
[the remark this answers is not in the document]
Comment (socratesgorilla) [reply python/doc/content/python/CSVDiff.md:1]: This version of csvdiff doesn't work as a standalone cmd tool

File: test/tests/test_harness/csvdiff_tests
Comment (socratesgorilla) [reply test/tests/test_harness/csvdiff_tests:1]: These tests are either for removed functionality or their testing has been moved to the new CSVDiff tests

File: python/TestHarness/testers/CSVDiff.py [hunk 3]
```diff
@@ -7,100 +7,57 @@
...
-
-            for command in commands:
-                exo_output = util.runCommand(command)
-                output += 'Running csvdiff: ' + command + '\n' + exo_output
-                if not "Files are the same" in exo_output:
-                    self.setStatus(self.diff, 'CSVDIFF')
-                    break
-
-        return output
+        return SchemaDiff.processResults(self, moose_dir, options, output)
```
[the remark this answers is not in the document]
Comment (socratesgorilla) [reply python/TestHarness/testers/CSVDiff.py]: Whoops, this was an artifact from development -- at first I was attempting to do more in the derived class. Removed it.

**Comment by GiudGiud** [comment]
where do we stand here?

**Comment by loganharbour** [comment]
We're standing.

Making sure that the error output is verbose enough compared to the old differ

**Comment by vanwdani** [comment]
Is this ready?  I'm hoping to use #23965 sooner or later

**Comment by permcody** [comment]
Closing in favor of #24633.

**Pull request closed without merging**